.github/workflows/nightly.yml

```diff
@@ -94,3 +94,65 @@ jobs:
       s3_bucket: ${{ secrets.IT_CASE_S3_BUCKET }}
       s3_access_key: ${{ secrets.IT_CASE_S3_ACCESS_KEY }}
       s3_secret_key: ${{ secrets.IT_CASE_S3_SECRET_KEY }}
+  prepare:
+    name: Prepare
+    runs-on: ubuntu-latest
+    outputs:
+      stringified-workflow-name: ${{ steps.workflow-prep-step.outputs.stringified-workflow-name }}
+    steps:
+      - name: "Stringify workflow name"
+        id: workflow-prep-step
+        run: |
+          # Similar to the template.flink-ci.yml compile job
+          stringified_workflow_name=$(echo "${{ github.workflow }}" | tr -C '[:alnum:]._' '-' |  tr '[:upper:]' '[:lower:]' | sed -e 's/--*/-/g' -e 's/^-*//g' -e 's/-*$//g')
+          echo "stringified-workflow-name=${stringified_workflow_name}" >> $GITHUB_OUTPUT
+  linux:
+    needs: prepare
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout the repository
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+          persist-credentials: false
+      - name: Install python
+        uses: actions/setup-python@v5
+        with:
+          python-version: ${{ env.PYTHON_VERSION }}
+      - name: Build wheels
+        run: |
+          cd flink-python
+          bash dev/build-wheels.sh
+      - name: Tar artifact
+        run: tar -czvf python-wheel.tar.gz flink-python/dist
+      - name: Upload artifact
+        uses: actions/upload-artifact@v4
+        with:
+          name: wheel_${{ needs.prepare.outputs.stringified-workflow-name }}-${{ github.run_number }}
```

Comment:
Shouldn't we add the `linux` phrase here somewhere? Otherwise, we would have the same build artifact name twice (for macos and linux). Or am I missing something?